lib/engine/game/g_1870/game.rb

```diff
@@ -791,8 +791,12 @@ def legal_tile_rotation?(_entity, hex, tile)
 
           (tile.exits & hex.tile.borders.select { |b| b.type == :water }.map(&:edge)).empty? &&
             hex.tile.partitions.all? do |partition|
-              tile.paths.all? do |path|
-                (path.exits - partition.inner).empty? || (path.exits - partition.outer).empty?
+              if partition.restrict != ''
```

Comment:
this seems strange to me, what if restrict is nil?

Reply:
It wasn't in my testing, but I can add a check.  Is there a shorthand way to test nil or blank?

Reply:
not in default ruby